### DISCUSSION TITLE
Public link lists

### DISCUSSION THREAD

**Discussion by erlend-sh** [raffomania/linkblocks#44] [Ideas]
My main use case for linkblocks is public link lists.

Example from [Linkding](https://github.com/sissbruecker/linkding): https://links.keybits.net/bookmarks/shared

Since public sharing would be my default, I'd prefer my links to be presented at `links.erlend.sh` or `erlend.sh/links`, whereas the private stuff is nested in a level deeper. Just like a blog that's default-public but also has a private section for drafts or restricted posts.

**Comment by raffomania** [comment]
Yes, this is something I very much want. The authentication system currently denies access to all pages, but once we have a boolean for setting a list to public or private, we can allow users without an account to see public lists. We'll need to think about the implications for bookmarks added to these lists - e.g. on a bookmark details page, only display public lists it's in.

In general, I'd like linkblocks' UI/UX to say "everything you do here is public by default". Private things are the exception.

**Reply by erlend-sh** [reply]
We've got a first-take of this in #52 so I'll make some inline threads below for further considerations in future work.

**Comment by erlend-sh** [comment]
### Distinct public/private views

public view should be distinct from private view. I.e. `links.erlend.sh/all` or `links.erlend.sh/lists/123` should display the same regardless of me being authenticated or not. When I view `blog.erlend.sh` I wanna read my blog like normal. I go to `blog.erlend.sh/ghost` to enter the backend.

**Reply by raffomania** [reply]
I'm not sure about this one, can you elaborate on your reasons for this? When I go to my github profile while authenticated, it's similar to to the logged in view but there are elements for customizing pins, etc. I like that.

**Reply by erlend-sh** [reply]
Yeah I’d still like some basic customization controls in the public view if I’m logged in. But I don’t want my authenticated state preventing me from seeing what my public page _actually_ looks like.

Right now, to confirm that the public view appears as I expect it to, I have to open it in incognito mode.

Gamedev analogy:
`links.erlend.sh/admin` = game editor view
`links.erlend.sh` (logged in) = live game with debug widgets
`links.erlend.sh` (anon / non-owner) = plain game view

**Reply by raffomania** [reply]
Oh right, github used to have a button for a "public view" of a profile as well. I think when you're authenticated, I'd expect linkblocks to redirect you to the private page from the index page, and then have a little "show public view" button somewhere.

**Comment by erlend-sh** [comment]
### Public all-view
A public all-view like `links.erlend.sh` or `links.erlend.sh/all `is more important than any specific list being public, as the all-view is the directory for all those other lists. How this should work may lack specification atm though.

**Comment by erlend-sh** [comment]
### Shorter list URLs

Currently a list URL looks as follows: `https://example.com/lists/119c98df-8d45-438e-b93f-b9a1fee36205`

Can we reduce the length of list URLs?

**Reply by raffomania** [reply]
I've toyed with the idea of using [nanoid](https://github.com/ai/nanoid). The obvious alternative would be to slugify the list name, which presents lots of interesting problems, like what to do when the name changes, or how to preserve the usefulness of slugs in the presence of chinese or arabic characters. Maybe using slugs by default, and allowing users to specify their own slug, is a good middle ground.
